enhancements/microshift/microshift-default-csi-plugin.md

```diff
@@ -0,0 +1,371 @@
+
+### Goals
+
+* Provide a default, production-ready, dynamic storage solution out of the box.
+* Protect the host OS from greedy workloads to assure system stability.
+* Align with MicroShift's lifecycle management model via rpm-ostree atomic updates.
+
+### Non-Goals
+
+* Support multi-node cluster topologies
+* Support remote storage volume provisioning and access
+
+## Proposal
+
+[TopoLVM](https://github.com/red-hat-storage/topolvm) (downstream, ODF-LVM), is the operand of the
+[LVM-Operator](https://github.com/red-hat-storage/lvm-operator), and is proposed as MicroShift's default supported
+CSI plugin. The storage solution is maintained downstream by the OpenShift Data Foundation (ODF) team, and is supported on
+Single Node OpenShift (SNO). The plugin is compatible with MicroShift's recommended OS's(RHEL/Centos/Fedora).
+
+ODF-LVM is production-ready, enables isolation of workload and system storage, and is already supported by ODF on Single
+Node OpenShift. ODF-LVM implements features not present in HPP, such as volume snapshotting and thin-provisioning.
+
+### Workflow Description
+
+> _Disclaimer:_ Workflows described here illustrate manual processes.  These are very similar to automated workflows we 
+> expect in production environements.  The primary difference is that users are not likely to have console access to 
+> the device. Instead, device owners will generate new OS layers containing the lvmd config and deploy these onto the
+> device.
+
+**Device Owner:** A human user with privileged access on the RHEL host and cluster-admin access in the MicroShift cluster.
+
+**Start MicroShift with default ODF-LVM configuration**
+
+> _Assuming a [RHEL 8.x](https://github.com/openshift/microshift/blob/main/docs/devenv_rhel8_auto.md) or 
+> [RHEL for Edge](https://github.com/openshift/microshift/blob/main/docs/rhel4edge_iso.md) host, ..._
+
+1. The device owner will connect to the device terminal.
+1. If the MicroShift service is not running, the device owner will start the service via `systemctl enable --now microshift`.
+1. The device owner will verify a stable deployment (see: **Validate-ODF-LVM function**)
+
+**Validate ODF-LVM function**
+1. The device owner will observe the cluster to confirm ODF-LVM pods are running with `oc get pods -n openshift-storage`.
+
+    ```shell
+    $ oc get pods -n openshift-storage
+    NAME                                  READY   STATUS    RESTARTS        AGE
+    topolvm-controller-8479455f95-tdv7f   4/4     Running   0               30m
+    topolvm-node-clv2z                    4/4     Running   0               30m
+    ```
+
+2. The device owner will create a simple Pod and PersistentVolumeClaim to test-drive the plugin with:
+    ```shell
+    cat <<'EOF' | oc apply -f -
+    kind: PersistentVolumeClaim
+    apiVersion: v1
+    metadata:
+      name: my-lv-pvc
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: 1G
```

Comment:
As no storageclass is provided, I assume that the topolvm storageclass which will be the default sc.

Reply:
That's correct.  The default deployment installs a single SC, marked as default.

Reply:
ODF LVMO configures LVM with thin provisioning.  Does Microshift intend to use lvm thin provisioning? The lvmd.yaml files in the PR will create thick lvs.

Reply:
Is the choice between thick or thin not available in LVMO?

Reply:
From 4.11 we mandate thin for ODF LVMO because we need snapshots and clones. Thick provisioning does not have those features. Topolvm supports both.

Reply:
IIRC, MicroShift is currently using the topolvm digests specified in the ocs-registry:[latest-4.10 or latest-stable-4.10] release image.  Rebase automation doesn't yet use this to get odf image digests for the corresponding OCP release, so the digests will have to be updated manually anyway.  We can bump it now unless there's a compelling reason not to.

Reply:
We've opened https://issues.redhat.com/browse/USHIFT-431 to make sure we document the recommended practices for using thin volumes.

@copejon , let's add this as a risk and say that our mitigation is that documentation.